File: backend/app/models/affirmation.py
from pydantic import BaseModel, Field

class AffirmationRequest(BaseModel):
    name: str = Field(..., min_length=1, max_length=50)
    feeling: str = Field(..., min_length=1, max_length=500)

class AffirmationResponse(BaseModel):
    affirmation: str

File: web-client/src/schemas/affirmation.ts
import { z } from "zod/v4";

export const affirmationRequestSchema = z.object({
	name: z
		.string()
		.min(1, "Name is required")
		.max(50, "Name must be 50 characters or less"),
	feeling: z
		.string()
		.min(1, "Please select or describe how you feel")
		.max(500, "Feeling must be 500 characters or less"),
});

export type AffirmationRequest = z.infer<typeof affirmationRequestSchema>;

export const affirmationResponseSchema = z.object({
	affirmation: z.string(),
});

export type AffirmationResponse = z.infer<typeof affirmationResponseSchema>;
